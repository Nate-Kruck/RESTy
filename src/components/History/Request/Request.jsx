import React from 'react';
import PropTypes from 'prop-types';

function Request({ method, url, onClick }) {
  return (
    <li onClick={() => onClick(url, method)}>
      <p>{method}</p>
      <p>{url}</p>
    </li>
  );
}

Request.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired
};

export default Request;
