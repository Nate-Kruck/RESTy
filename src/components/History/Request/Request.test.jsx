import React from 'react';
import {
  render,
  cleanup
} from '@testing-library/react';
import Request from './Request';

describe('Request Testing', () => {
  afterEach(() => cleanup());

  it('Makes a request onClick', () => {
    const { asFragment } = render(<Request
      method="method"
      url="url"
      onClick={() => {}}
    />);

    expect(asFragment()).toMatchSnapshot();
  });
});
