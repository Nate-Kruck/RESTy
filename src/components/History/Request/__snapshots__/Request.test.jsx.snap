// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Request Testing Makes a request onClick 1`] = `
<DocumentFragment>
  <li>
    <p>
      method
    </p>
    <p>
      url
    </p>
  </li>
</DocumentFragment>
`;
